test/unit/test_metrics.py

```diff
@@ -0,0 +1,159 @@
+from copy import deepcopy
+
+import pytest
+
+from curvesim.exceptions import StateLogError
+from curvesim.metrics import init_metrics, metrics
+from curvesim.metrics.base import PoolMetric, PricingMetric
+from curvesim.metrics.state_log.pool_parameters import (
+    get_pool_parameters,
+    pool_parameter_functions,
+)
+from curvesim.metrics.state_log.pool_state import get_pool_state, pool_state_functions
+
+all_metrics = [getattr(metrics, m) for m in metrics.__all__]
+
+
+def test_get_pool_parameters(
+    sim_curve_pool, sim_curve_crypto_pool, sim_curve_meta_pool, sim_curve_rai_pool
+):
+    """Test that get_pool_parameters returns the right data for each pool type."""
+    pools = [
+        sim_curve_pool,
+        sim_curve_crypto_pool,
+        sim_curve_meta_pool,
+        sim_curve_rai_pool,
+    ]
+
+    for pool in pools:
+        params = get_pool_parameters(pool)
+        assert params == pool_parameter_functions[type(pool)](pool)
+
+
+def test_get_pool_state(
+    sim_curve_pool, sim_curve_crypto_pool, sim_curve_meta_pool, sim_curve_rai_pool
+):
+    """Test that get_pool_parameters returns the right data for each pool type."""
+    pools = [
+        sim_curve_pool,
+        sim_curve_crypto_pool,
+        sim_curve_meta_pool,
+        sim_curve_rai_pool,
+    ]
+
+    for pool in pools:
+        state = get_pool_state(pool)
+        assert state == pool_state_functions[type(pool)](pool)
+
+
+def test_get_pool_parameters_unmapped_exception():
+    """Test that unmapped pool throws StateLogError."""
+
+    class DummyPool:
+        pass
+
+    pool = DummyPool()
+
+    with pytest.raises(StateLogError):
```

Comment:
would probably be better to raise a more specific error...

class UnregisteredPoolError(StateLogError):
    pass

Reply:
Good call. Changed